src/Microsoft.Diagnostics.Runtime/src/Windows/CachedMemoryReader.cs

```diff
@@ -85,7 +83,13 @@ public CachedMemoryReader(ImmutableArray<MinidumpSegment> segments, string dumpP
             else
             {
                 // We can't add the lock memory privilege, so just fall back on our ArrayPool/MemoryMappedFile based cache 
-                _cachedMemorySegments = new HeapSegmentDataCache(new ArrayPoolBasedCacheEntryFactory(stream, leaveOpen), MaxCacheSize);
+
+                // Force creation of emtpy entries for each segment, this won't map the data in from disk but it WILL prevent us from needing to take any locks at the first level of the cache
+                // (the individual entries, when asked for data requiring page-in or when evicting data in page-out will still take locks for consistency).
+                foreach (MinidumpSegment segment in _segments)
+                    _cachedMemorySegments.CreateAndAddEntry(segment);
```

Comment:
You use `_cachedMemorySegments` before you assign it here.

Reply:
Doh, I literally added that just in the CLRMD code port, shows me for trying to add a untested improvement in a PR :(

Reply:
Let me flip the order